rfd/0072-ec2-tags.md

```diff
@@ -26,21 +29,36 @@ As of January 2022, [instance tags are available via the instance metadata servi
+When a Teleport process running an SSH or other service is started, check if it is running in a cloud instance. If it is, fetch all tags from the instance metadata service and add them as labels to Teleport, then start a service that periodically (every hour) that does the same. Updates will replace all labels from the previous iteration, so newly created or deleted tags will be reflected in Teleport.
 
-EC2 labels will use the `aws` prefix in Teleport for namespacing (similar to `teleport.dev`). For example, if the instance has the tag `Name: my-instance`, Teleport will add the label `aws/Name: my-instance`. Ideally, the `aws` namespace will prevent collisions with static and command label names, but if there is a collision, EC2 labels will not override static or command labels.
+Cloud labels will use a namespace prefix in Teleport for namespacing (similar to `teleport.dev`). For example, if an EC2 instance has the tag `Name: my-instance`, Teleport will add the label `aws/Name: my-instance`. Ideally, the namespace will prevent collisions with static and command label names, but if there is a collision, cloud labels will not override static or command labels.
+
+### EC2
 
 In order to use this feature, instance tags in metadata must be enabled for the instance. Instance tags in metadata can be enabled/disabled when launching a new instance; they can also be toggled for an existing instance via `Actions > Instance settings > Allow tags in instance metadata` in the management console or with the `modify-instance-metadata-options` command in the AWS CLI. See the [AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#allow-access-to-tags-in-IMDS) for more details.
 
 Note: Some instance types (specifically non-[Nitro](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances)) [do not propagate tag updates until after a restart](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html#work-with-tags-in-IMDS). This will need to be mentioned in the documentation.
 
-### Throttling
+Namespace prefix: `aws`
+
+#### Throttling
+
 AWS applies [per-instance throttling to instance metadata requests](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html#instancedata-throttling). Each instance can have at most 50 tags, so with one request to fetch all tag keys and up to 50 requests to fetch tag values, Teleport only needs to make at most 51 instance metadata requests per hour, which should not have throttling issues.
 
+### Azure
+
+Azure instance metadata (including tags) is always enabled and requires no additional setup from the user. Azure VMs do not require a restart for tag updates to apply.
```

Comment:
Always enabled means that it is enabled and cannot be disabled, or is it enabled by default, and there is a way to disable it?
I'd assume the first, but I want to double check.

Reply:
It can't be disabled.